### PR TITLE
console.warn when an error is encountered in a result reducer

### PR DESCRIPTION
**Pull request by BlooJeans**
TODO:

- [ ] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [ ] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change
- [x] Add your name and email to the AUTHORS file (optional)
- [ ] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Expect active development and potentially significant breaking changes in the `0
 - Fix: make sure maybeDeepFreeze is called on results returned from setVariables and refetch [PR #1362](https://github.com/apollographql/apollo-client/pull/1362)
 - Fix: use setTimeout to throw uncaught errors in observer.next and observer.error[PR #1367](https://github.com/apollographql/apollo-client/pull/1367)
 - Remove returnPartialData option [PR #1370](https://github.com/apollographql/apollo-client/pull/1370)
+- console.warn() when an exception is encountered in a result reducer
```
Comment (helfer) [inline CHANGELOG.md]: Make sure to add the PR number here.

File: src/data/resultReducers.ts [hunk 2]
```diff
@@ -56,8 +56,13 @@ export function createStoreReducer(
       return store;
     }
 
-
-    const nextResult = resultReducer(result, action, variables); // action should include operation name
+    let nextResult;
+    try {
+      nextResult = resultReducer(currentResult, action, variables); // action should include operation name
```
Comment (helfer) [inline src/data/resultReducers.ts]: currentResult is not defined.

File: src/data/resultReducers.ts [hunk 3]
```diff
@@ -56,8 +56,13 @@ export function createStoreReducer(
...
+    } catch (err) {
+      console.warn('Error in a resultReducer, ignoring and moving on.  ', err);
```
Comment (helfer) [inline src/data/resultReducers.ts]: I think just "Unhandled error in result reducer" would be better.

File: src/data/resultReducers.ts (hunk 2 above)
Comment (BlooJeans) [inline src/data/resultReducers.ts]: whoops, sorry - looks like the variable name changed between 0.8.7 and origin/master

**Comment by BlooJeans** [comment]
All tests passing - I had to add the console.warn reassign to the appropriate test to show that it's purposefully emitting a warning.  I did it inline at first but then rewrote it to use the describe() block syntax - seems safer, but slightly noisier.  If you prefer it the original inline way I can revert it.

**Comment by helfer** [comment]
Looks good to me. Thanks a lot @BlooJeans ! 👍

**Pull request merged**